ambry-router/src/main/java/com/github/ambry/router/OperationController.java

```diff
@@ -106,13 +107,15 @@ public OperationController(String suffix, String defaultPartitionClass, AccountS
     this.responseHandler = responseHandler;
     this.nonBlockingRouter = nonBlockingRouter;
     // Warm up connections to dataNodes in local and remote DCs.
-    List<ResponseInfo> responseInfos = Collections.synchronizedList(new ArrayList<>());
```

Comment:
What changed that is causing this exception now ?

Reply:
This seems to have been present always. We only see if we return from `networkClient.warmUpConnections` on timeout (5 seconds) and the `responseInfo` List is updated by Netty threads as we try to iterate over it. Due to that, the probability might have been less.